interface/reports/patient_list_creation.php

```diff
@@ -246,582 +319,773 @@ function srch_option_change(elem) {
+                                                    value="<?php echo attr($skey); ?>"><?php echo text(xl($skey)); ?></option>
+                                                <?php } ?>
+                                            </select>
+                                        </td>
+                                        <td colspan="2">
+                                            <!-- Inputs for specific search options -->
+                                            <span id="rx_drug" style="display: none">
+                                                <input class="form-control wildcard_field" name="prescription_drug" id="prescription_drug" title="<?php echo xla('(% matches any string, _ matches any character)'); ?>" placeholder="<?php echo xlt('Drug'); ?>"<?php echo !empty($_POST['prescription_drug']) ? ' value="' . $_POST['prescription_drug'] . '"' : '' ?>/>
+                                            </span>
+                                            <span id="com_pref" style="display: none">
+                                                <select class="form-control" name="communication" id="communication" title="<?php echo xlt('Select Communication Preferences'); ?>">
+                                                    <option><?php echo xlt('All'); ?></option>
+                                                    <option value="allow_sms" <?php echo ($communication == "allow_sms") ? "selected" : ""; ?>><?php echo xlt('Allow SMS'); ?></option>
+                                                    <option value="allow_voice" <?php echo ($communication == "allow_voice") ? "selected" : ""; ?>><?php echo xlt('Allow Voice Message'); ?></option>
+                                                    <option value="allow_mail" <?php echo ($communication == "allow_mail") ? "selected" : ""; ?>><?php echo xlt('Allow Mail Message'); ?></option>
+                                                    <option value="allow_email" <?php echo ($communication == "allow_email") ? "selected" : ""; ?>><?php echo xlt('Allow Email'); ?></option>
+                                                </select>
+                                            </span>
+                                            <span id="ins_co" style="display: none">
+                                                <select class="form-control" name="insurance_companies" id="insurance_companies" title="<?php echo xlt('Select Insurance Company'); ?>">
+                                                    <option><?php echo xlt('All'); ?></option>
+                                                    <?php foreach ($insarr as $ins_id => $ins_co) { ?>
+                                                        <option <?php echo (!empty($_POST['insurance_companies']) && ($_POST['insurance_companies'] == $ins_id)) ? 'selected' : ''; ?> value="<?php echo $ins_id; ?>"><?php echo text($ins_co); ?></option>
+                                                    <?php } ?>
```

Comment:
$ins_id needs to be wrapped with attr() function for html escaping (prevents cross-scripting attacks)